### ISSUE TITLE
Make the mail example imap4client.py compatible with servers require SSL

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5697]
|[<img alt="argonemyth's avatar" src="https://avatars.githubusercontent.com/u/965739?s=50" width="50" height="50">](https://github.com/argonemyth)| @argonemyth reported|
|-|-|
|Trac ID|trac#5697|
|Type|enhancement|
|Created|2012-05-31 16:14:48Z|
|Branch|https://github.com/twisted/twisted/tree/imap-example-5697|

The example only works with an IMAP server that doesn't require SSL. The script hangs if you try to access servers like imap.gmail.com. I guess a lot of people might try Gmail with this script, it's better to make it compatible with servers that require SSL. 

The script should also stop the reactor once the job is finished.

Attachments:

* [5697-imap4client-20120612.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/53a/53a5e159ca63c196633afa43976d1cba5644f38c/2649d93d572c93f43c8a41fb7de9e8ebddb29ed4.patch) (7149 bytes) - added by argonemyth on 2012-06-01 17:07:57Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5697 5697
type__enhancement enhancement
reporter__argonemyth argonemyth
priority__normal normal
milestone__ 
branch__branches_imap_example_5697 branches/imap-example-5697
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__documentation__example documentation, example
time__1338480888000000 1338480888000000
changetime__1474197555542286 1474197555542286
version__None None
owner__itamar itamar
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="argonemyth's avatar" src="https://avatars.githubusercontent.com/u/965739?s=50" width="50" height="50">](https://github.com/argonemyth)<a name="note_1"></a>|@argonemyth removed owner|
|-|-|

The main reason why it hangs when you try to connect to a Gmail account is that it doesn't initiate a SSL connection if you only change the port number to 993. I fixed the issue by using **reactor.connectSSL** if an user input '993' as the port number.

A note on Gmail! It's been using 'XOAUTH' as authentication method since 2010. So, only 'clear-text' (the insecure way) is available for gmail accounts. I tested the script on an IMAP server that supports PLAIN auth, worked well. 

I also modified the example a bit to fit the coding standard. Please review (5697-imap4client-20120612.patch), Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst set owner to @itamarst|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst commented|
|-|-|

(In [34752]) Branching to 'imap-example-5697'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [34756]) Merge imap-example-5697: Improvements to the IMAP client example.

Author: argonemyth
Reviewer: itamar
Fixes: [#5697](https://github.com/twisted/twisted/issues/5697)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
